main.tf

```diff
@@ -1,13 +1,76 @@
-terraform {
-  required_version = ">= 1.3.0"
-  required_providers {
-    aws = {
-      source  = "hashicorp/aws"
-      version = ">= 4.0"
+locals {
+  runner_run_as = "runners"
+
+  runner_base_config = { # Base configuration shared between arm and amd runners
+    fifo                = true
+    delay_webhook_event = 0
+
+    runner_config = {
+      create_service_linked_role_spot     = true
+      instance_allocation_strategy        = var.instance_allocation_strategy
+      runners_maximum_count               = var.runners_maximum_count
+      enable_organization_runners         = true
+      runner_group_name                   = var.runner_group_name
+      runner_iam_role_managed_policy_arns = var.runner_iam_role_managed_policy_arns
+      runner_os                           = "linux"
+      runner_extra_labels                 = "ondemand"
+      enable_ssm_on_runners               = true
+      idle_config                         = var.idle_config
+      runner_run_as                       = local.runner_run_as
+      userdata_template                   = "${path.module}/templates/user_data.sh"
+      userdata_pre_install                = var.userdata_pre_install
+      userdata_post_install               = var.userdata_post_install
+      ami_owners                          = ["137112412989"] # amazon
+
+      block_device_mappings = [
+        {
+          device_name           = "/dev/xvda"
+          delete_on_termination = true
+          volume_type           = "gp3"
+          volume_size           = var.volume_size
+          iops                  = null
+          encrypted             = false
+          kms_key_id            = null
+          snapshot_id           = null
+          throughput            = null
+        }
+      ]
+
+      runner_log_files = var.runner_log_files
     }
   }
-}
 
-provider "aws" {
-  region = "eu-central-1"
+  amd_runner = var.deploy_amd ? {
+    "linux-x64" = merge(local.runner_base_config, { # merge of architecture dependant configuration
+      runner_config = merge(local.runner_base_config.runner_config, {
+        runner_architecture = "x64"
+        instance_types      = var.amd_instance_types
+      })
+      matcherConfig = {
+        labelMatchers = [concat(["x64"], var.runners_labels)]
+        exactMatch    = true
+      }
+      ami_filter = {
+        name = ["amzn2-ami-kernel-5.*-hvm-*-x86_64-gp2"]
+      }
+    })
+  } : {}
+
+  arm_runner = var.deploy_arm ? {
+    "linux-arm64" = merge(local.runner_base_config, { # merge of architecture dependant configuration
+      runner_config = merge(local.runner_base_config.runner_config, {
+        runner_architecture = "arm64"
+        instance_types      = var.arm_instance_types
+      })
+      matcherConfig = {
+        labelMatchers = [concat(["arm64"], var.runners_labels)]
+        exactMatch    = true
+      }
+      ami_filter = {
+        name = ["amzn2-ami-kernel-5.*-hvm-*-amd64-gp2"]
```

Comment:
@DCamma Is this the right AMI tag? 

I think you mis-copied it. Have a look at the current one in Staging https://github.com/DND-IT/DND-AWS-on-demand-github-runner/blob/main/main.tf#LL154